server/config/main.go

```diff
@@ -0,0 +1,70 @@
+package config
```

Comment:
1. Using a separate package for `config` makes the config available for import in other packages without the need for circular import.
`api` and `command` packages are imported by the `main` package. If `config` is a part of the `main` package, we wouldn't be able to use `config` in those two packages.

2. I think using `atomic.Value` is much cleaner way to maintain the config than using locks. It might even be faster. Refer: https://medium.com/a-journey-with-go/go-how-to-reduce-lock-contention-with-the-atomic-package-ba3b2664b549

Reply:
On second thought, the more go-like way of doing this might be to use `channels` for updating the in-memory configuration.
This might still need some more thought, but roughly, the implementation could work like this:
1. In plugin `OnActivate`
    - Initialize the `channel` to update config and store that in the `Plugin` struct.
    - Start the goroutine that will update the config.

2. On every `OnConfigurationChange`
    - Use `p.API.LoadPluginConfiguration` and pass that to the the config-update `channel`.

I do, however, think that using `atomic.Value` would be a much simpler approach.
Thoughts?